### PR TITLE
M0078DDG-400 Update CI bundle to use MMM CLI and remove MicroEJ git flow scripts

### PR DESCRIPTION
**Pull request by tdelhomenie-microej**

### CODE REVIEW

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 1]
```diff
@@ -48,13 +48,15 @@ The overall build and deployment flow of a module can be summarized as follows:
...
-*  `MicroEJ SDK <https://developer.microej.com/get-started/>`_ ``4.1.5`` or higher.
+*  `MicroEJ SDK <https://developer.microej.com/get-started/>`_ ``5.4.0`` or higher.
 *  Git ``2.x`` installed, with Git executable in path. We recommend installing Git Bash if your operating system is Windows (`<https://git-for-windows.github.io/>`_).
-*  Apache Ant ``1.9.x`` installed (`<https://ant.apache.org/bindownload.cgi>`_).
 *  Java Development Kit (JDK) ``1.8.x``.
 
 This tutorial was tested with Jenkins ``2.235.3`` and Artifactory ``6.20.1``.
 
+.. note::
+    For SDK versions before 5.4.0, please refer to :ref:`this tutorial <tutorial_setup_automated_build_using_jenkins_and_artifactory_up_to_sdk_5_3>`.
```
Comment (frivieremicroej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:58]: Use the "PDF pattern" instead as done here: https://github.com/MicroEJ/docs/blame/master/ApplicationDeveloperGuide/UI/ui.rst#L19

=> 
please refer to this `MicroEJ Documentation Archive <https://docs.microej.com/_/downloads/en/20210521/pdf/>`_. 

(the link is available)

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 2]
```diff
@@ -68,53 +70,47 @@ The steps to follow are:
...
-
-   .. code-block:: sh
-   
-      git clone --recursive https://github.com/MicroEJ/Tool-CommandLineBuild.git
-
-#. Create a directory named ``buildKit`` in the ``Tool-CommandLineBuild`` directory.
+#. Create a directory named ``buildKit``.
 #. Export the MicroEJ build kit from your MicroEJ SDK version to the ``buildKit`` directory, by following the steps described :ref:`here <mmm_build_kit>`.
-#. Set the environment variable ``MICROEJ_BUILD_TOOLS_HOME`` to point to the ``Tool-CommandLineBuild`` directory
+#. Download :download:`this Ivy settings file <resources/ivysettings-artifactory.xml>` and save it at the root of the ``buildKit`` directory. This file pre-configures Ivy to fetch and publish modules from the Artifactory server used in this tutorial.
```
Comment (frivieremicroej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:92]:
```suggestion
#. Download :download:`this settings file <resources/ivysettings-artifactory.xml>` and save it at the root of the ``buildKit`` directory. This file configures MicroEJ Module Manager to import and publish modules from the Artifactory repositories described in this tutorial. Please refer to <settings file> section for more details.

```

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 3]
```diff
@@ -230,29 +220,30 @@ Install Jenkins
...
-    #. Set :guilabel:`Name` to ``Ant 1.9``.
-    #. Uncheck :guilabel:`Install automatically`.
-    #. Set :guilabel:`ANT_HOME` to ``path/to/apache-ant-1.9.[version]``.
+    #. Set :guilabel:`JAVA_HOME` to the absolute path of your JDK installation (for example ``C:\Program Files\Java\jdk1.8.0_[version]`` on Windows).
+#. Click on :guilabel:`Save`.
+#. Go to :guilabel:`Manage Jenkins` > :guilabel:`Configure System`.
+    #. Scroll to :guilabel:`Global properties` section.
+    #. Check :guilabel:`Environment variables`.
+    #. Click on :guilabel:`Add`.
+    #. Set :guilabel:`Name` to ``MICROEJ_BUILD_TOOLS_HOME``.
```
Comment (frivieremicroej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:245]:
```suggestion
    #. Set :guilabel:`Name` to ``MICROEJ_BUILD_KIT_HOME``.
```

File: Tutorials/resources/ivysettings-artifactory.xml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+	<!-- Fetch resolver that looks everywhere to get the artifacts -->
+	<chain name="fetchAll">
+		<resolver ref="fetchPersonal"/>
+	</chain>
+	<ibiblio name="ext" m2compatible="true" root="${Artifactory_Repository}/ext" />
+	
+  </resolvers>
+  
+  <modules>
```
Comment (frivieremicroej) [inline Tutorials/resources/ivysettings-artifactory.xml:132]: I think this section can be fully removed as it it build repository specific (declared in MMM build repository settings  file)

File: Tutorials/resources/ivysettings-artifactory.xml [hunk 5]
```diff
@@ -0,0 +1,148 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (frivieremicroej) [inline Tutorials/resources/ivysettings-artifactory.xml:1]: Try to reduce this settings file to the minimum resolvers necessary to run this tutorial

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 6]
```diff
@@ -68,53 +70,47 @@ The steps to follow are:
...
 
-For the purposes of simplifying the steps, this tutorial will be performed locally on a single machine.
+In order to simplify the steps, this tutorial will be performed locally on a single machine.
 
-Artifactory will host MicroEJ modules in 3 repositories:
+Artifactory will host MicroEJ modules in 2 repositories:
 
 - ``microej-module-repository``: repository initialized with pre-built MicroEJ modules, a mirror of the :ref:`Central Repository <central_repository>`
-- ``microej-build-repository``: repository initialized with build scripts and tools exported from MicroEJ SDK
 - ``libs-snapshot-local``: repository where custom modules will be published
```
Comment (frivieremicroej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:78]: rename libs to modules => ``modules-snapshot-local``

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 7]
```diff
@@ -68,53 +70,47 @@ The steps to follow are:
...
-
-   .. code-block:: sh
-   
-      git clone --recursive https://github.com/MicroEJ/Tool-CommandLineBuild.git
-
-#. Create a directory named ``buildKit`` in the ``Tool-CommandLineBuild`` directory.
+#. Create a directory named ``buildKit``.
 #. Export the MicroEJ build kit from your MicroEJ SDK version to the ``buildKit`` directory, by following the steps described :ref:`here <mmm_build_kit>`.
-#. Set the environment variable ``MICROEJ_BUILD_TOOLS_HOME`` to point to the ``Tool-CommandLineBuild`` directory
+#. Download :download:`this Ivy settings file <resources/ivysettings-artifactory.xml>` and save it at the root of the ``buildKit`` directory. This file pre-configures Ivy to fetch and publish modules from the Artifactory server used in this tutorial.
```
Comment (frivieremicroej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:91]: This does not work for PDF 
image

=> I suggest to inline the settings file (assuming it has been strongly simplified)

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 6 above)
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:78]: `libs-snapshot-local` is one of the default repositories of Artifactory, so we have to add a step to create a new repository.

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 1 above)
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst]: Done

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 2 above)
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst]: Done

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 3 above)
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst]: Done

File: Tutorials/resources/ivysettings-artifactory.xml (hunk 4 above)
Comment (tdelhomenie-microej) [reply Tutorials/resources/ivysettings-artifactory.xml]: Done

File: Tutorials/resources/ivysettings-artifactory.xml (hunk 5 above)
Comment (tdelhomenie-microej) [reply Tutorials/resources/ivysettings-artifactory.xml:1]: Done.
I am just wondering if we keep the ibiblio resolver (maven). We should either use it or remove it. WDYT ?

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 7 above)
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst]: Done. Content inlined.

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 8]
```diff
@@ -48,13 +48,15 @@ The overall build and deployment flow of a module can be summarized as follows:
 Prerequisites
 -------------
 
-*  `MicroEJ SDK <https://developer.microej.com/get-started/>`_ ``4.1.5`` or higher.
+*  `MicroEJ SDK <https://developer.microej.com/get-started/>`_ ``5.4.0`` or higher.
 *  Git ``2.x`` installed, with Git executable in path. We recommend installing Git Bash if your operating system is Windows (`<https://git-for-windows.github.io/>`_).
-*  Apache Ant ``1.9.x`` installed (`<https://ant.apache.org/bindownload.cgi>`_).
 *  Java Development Kit (JDK) ``1.8.x``.
 
 This tutorial was tested with Jenkins ``2.235.3`` and Artifactory ``6.20.1``.
```
Comment (tdelhomenie-microej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:55]: @frivieremicroej I tested with newer versions (Jenkins 2.277.4 and Artifactory 6.23.13), so I suggest to update them in the tuto. WDYT ?

**Comment by frivieremicroej** [comment]
https://docs.microej.com/en/feature-m0078ddg-400_ci-bundle-tuto/

File: Tutorials/resources/ivysettings-artifactory.xml (hunk 5 above)
Comment (frivieremicroej) [reply Tutorials/resources/ivysettings-artifactory.xml:1]: If it works without ibibilio, remove it

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 8 above)
Comment (frivieremicroej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:55]: OK

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 6 above)
Comment (frivieremicroej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:78]: OK now I remember => so let it as is

File: Tutorials/resources/ivysettings-artifactory.xml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+<ivy-settings>
```
Comment (frivieremicroej) [inline Tutorials/resources/ivysettings-artifactory.xml:2]: Check if we can simplifiy like this:
- removed settings file comments to reduce the inline file. Moreover, comments were no so up-to-date (product beta...)
- removed `lib-release` useless ?
- removed `fetchAll` useless ?
- removed ibiblio
```
<resolvers>
 
	<url name="publishRelease" m2compatible="true">
		<artifact pattern="${Artifactory_Repository}/libs-release-local/[organization]/[module]/[branch]/[revision]/[artifact]-[revision](-[classifier]).[ext]" />
		<ivy pattern="${Artifactory_Repository}/libs-release-local/[organization]/[module]/[branch]/[revision]/ivy-[revision].xml" />
	</url>
	
	<url name="publishSnapshot" m2compatible="true" checkmodified="true">
		<artifact pattern="${Artifactory_Repository}/libs-snapshot-local/[organization]/[module]/[branch]/[revision]/[artifact]-[revision](-[classifier]).[ext]" />
		<ivy pattern="${Artifactory_Repository}/libs-snapshot-local/[organization]/[module]/[branch]/[revision]/ivy-[revision].xml" />
	</url>
	
	<filesystem name="publishPersonal" m2compatible="true" checkmodified="true">
		<artifact pattern="${personalBuilds.dir}/[organization]/[module]/[branch]/[revision]/[artifact]-[revision](-[classifier]).[ext]" />
		<ivy pattern="${personalBuilds.dir}/[organization]/[module]/[branch]/[revision]/ivy-[revision].xml" />
	</filesystem>

	<url name="fetchMicroEJModules" m2compatible="true">
		<artifact pattern="${Artifactory_Repository}/microej-module-repository/[organization]/[module]/[branch]/[revision]/[artifact]-[revision](-[classifier]).[ext]" />
		<ivy pattern="${Artifactory_Repository}/microej-module-repository/[organization]/[module]/[branch]/[revision]/ivy-[revision].xml" />
	</url>

	<chain name="fetchRelease">
		<resolver ref="publishRelease"/>
		<resolver ref="fetchMicroEJModules"/>
	</chain>
	
	<chain name="fetchSnapshot">
		<resolver ref="publishSnapshot"/>
		<resolver ref="fetchRelease"/>
	</chain>

	<chain name="fetchPersonal">
		<resolver ref="publishPersonal"/> 
		<resolver ref="fetchSnapshot"/> 
	</chain>
  </resolvers>
```

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 10]
```diff
@@ -68,53 +70,52 @@ The steps to follow are:
...
-    │   ├── ivysettings-artifactory.xml
+    ├── microej-build-repository
+    │   ├── ant-contrib
+    │   ├── be
     │   └── ...
-    └── ...
+    ├── microej-module-repository
+    │   └── ivysettings.xml
+    ├── release.properties
+    └── ivysettings-artifactory.xml
```
[the remark this answers is not in the document]
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:118]: I guess it has been done like that initially to make it easier to update the build kit (everything can be replaced).

File: Tutorials/resources/ivysettings-artifactory.xml (hunk 9 above)
Comment (tdelhomenie-microej) [reply Tutorials/resources/ivysettings-artifactory.xml:2]: > removed settings file comments to reduce the inline file. Moreover, comments were no so up-to-date (product beta...)

Comments can help to understand how things work. Maybe we should update them to be more relevant ?

> removed lib-release useless ?

libs-release is a default Artifactory (virtual) repository which gathers the 2 repositories libs-release-local and jcenter. So it allows to fetch releases built internally but also fetch public modules.

> removed fetchAll useless ?

OK

> removed biblio

OK

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 8 above)
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst]: Done.

File: Tutorials/resources/ivysettings-artifactory.xml (hunk 5 above)
Comment (tdelhomenie-microej) [reply Tutorials/resources/ivysettings-artifactory.xml:1]: Done

File: Tutorials/resources/ivysettings-artifactory.xml (hunk 9 above)
Comment (tdelhomenie-microej) [reply Tutorials/resources/ivysettings-artifactory.xml:2]: Done

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:118]: OK Edit the file and replace with following settings for artifactory...

File: Tutorials/resources/ivysettings-artifactory.xml [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+		<resolver ref="publishSnapshot"/>
+		<resolver ref="fetchRelease"/>
+	</chain>
+
+	<!-- Fetch resolver for personal build: look in personal, snapshot and release repositories -->
+	<chain name="fetchPersonal">
+		<resolver ref="publishPersonal"/> 
+		<resolver ref="fetchSnapshot"/> 
+	</chain>
+
```
Comment (frivieremicroej) [inline Tutorials/resources/ivysettings-artifactory.xml:76]: add fetchAll alias to fetchLocal

File: Tutorials/resources/ivysettings-artifactory.xml (hunk 9 above)
Comment (frivieremicroej) [reply Tutorials/resources/ivysettings-artifactory.xml:2]: remove useless lib-release and jcenter

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst [hunk 12]
```diff
@@ -183,12 +184,10 @@ Next step is to create the repositories that will hold the MicroEJ modules.
...
-#. Set :guilabel:`Repository Key` field to ``microej-build-repository``, then uncheck :guilabel:`Handle Snapshots`. Click on :guilabel:`Save and finish`.
-#. Make these two repositories accessible by default:
+#. Make this repository accessible by default:
     #. Go to :guilabel:`Admin` > :guilabel:`Security` > :guilabel:`Permissions`. 
     #. Click on :guilabel:`Anything` entry (do not check the line)
-    #. On the :guilabel:`Resources` tab, drag repositories ``microej-module-repository`` and ``microej-build-repository`` from the :guilabel:`Available repositories` area to the :guilabel:`Included Repositories` area.
+    #. On the :guilabel:`Resources` tab, drag repository ``microej-module-repository`` from the :guilabel:`Available repositories` area to the :guilabel:`Included Repositories` area.
     #. Click on :guilabel:`Save & Finish`.
 
 .. image:: images/tuto_microej_cli_artifactory_permissions.PNG
```
Comment (fpinel-microej) [inline Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:193]: This image still mentions microej-build-repository, which got me confused for a second

**Comment by fpinel-microej** [comment]
l.428 `#. Set :guilabel:`Test report XMLs` to ``**/target~/test/xml/**/test-report.xml, **/target~/test/xml/**/*Test.xml``.` still gives me the`'**/target~/test/xml/**/test-report.xml' ne correspond à rien : '**' existe mais pas '**/target~/test/xml/**/test-report.xml'` error as it did the first time I followed this tutorial

**Comment by fpinel-microej** [comment]
Note : l.228 On Chrome, the link to download `jenkins.war` would not work (the tab it opened would close directly without anything download starting) without any warning from the browser. I had to find another way to trigger the download for Chrome to finally warn me that the download was being blocked because the file was suspicious, so I could manually allow downloading.

**Comment by frivieremicroej** [comment]
> l.428 `#. Set :guilabel:`Test report XMLs`` to ``**/target~/test/xml/**/test-report.xml, **/target~/test/xml/**/*Test.xml``. `` still gives me the`'**/target~/test/xml/**/test-report.xml' ne correspond à rien : '**' existe mais pas '**/target~/test/xml/**/test-report.xml'` error as it did the first time I followed this tutorial

Yes, @tdelhomenie-microej please add a note to indicate this error is normal because the folders will be generated later

**Comment by frivieremicroej** [comment]
> Note : l.228 On Chrome, the link to download `jenkins.war` would not work (the tab it opened would close directly without anything download starting) without any warning from the browser. I had to find another way to trigger the download for Chrome to finally warn me that the download was being blocked because the file was suspicious, so I could manually allow downloading.

@tdelhomenie-microej maybe also add the wget command line ?

File: Tutorials/resources/ivysettings-artifactory.xml [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+	<property name="artifactory.repository.url" value="http://localhost:8081/artifactory" override="false"/>
+	<property name="local.repository.dir" value="${user.home}/.ivy2/repository/" override="false"/>
+  
+	<!--
+		Map MMM resolvers (*.resolver) to custom resolver
+		
+		Kinds of repositories:
+		- release: used when releasing a module.
+		- snapshot: used when building a module snapshot.
+		- personal: used when building a module in local or with a personal build.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply Tutorials/resources/ivysettings-artifactory.xml]: @tdelhomenie-microej  is there a reason that suggestion has not been applied as-is (missing personal renamed to local)

File: Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst (hunk 12 above)
Comment (tdelhomenie-microej) [reply Tutorials/tutorialSetupBuildUsingJenkinsAndArtifactory.rst:193]: Image updated.

**Comment by tdelhomenie-microej** [comment]
> 
> 
> l.428 `#. Set :guilabel:`Test report XMLs`` to ``**/target~/test/xml/**/test-report.xml, **/target~/test/xml/**/*Test.xml``. `` still gives me the`'**/target~/test/xml/**/test-report.xml' ne correspond à rien : '**' existe mais pas '**/target~/test/xml/**/test-report.xml'` error as it did the first time I followed this tutorial

Note added.

**Comment by tdelhomenie-microej** [comment]
> 
> 
> > Note : l.228 On Chrome, the link to download `jenkins.war` would not work (the tab it opened would close directly without anything download starting) without any warning from the browser. I had to find another way to trigger the download for Chrome to finally warn me that the download was being blocked because the file was suspicious, so I could manually allow downloading.
> 
> @tdelhomenie-microej maybe also add the wget command line ?

I change the link to point to the download page of Jenkins, it is more generic, and would allow customers to choose others packages if they prefer (I met this case with a customer last week).

**Pull request merged**